src/js/components/RangeInput/RangeInput.js

```diff
@@ -1,40 +1,117 @@
 import React, { forwardRef, useContext, useState } from 'react';
-
+import { Box } from '../Box';
+import { TextInput } from '../TextInput';
 import { FormContext } from '../Form/FormContext';
 import { StyledRangeInput } from './StyledRangeInput';
 
+const allowedKeysToOpenInputMask = [
+  'ArrowRight',
+  'ArrowLeft',
+  'ArrowUp',
+  'ArrowDown',
+];
+const allowedKeysToCloseInputMask = ['Tab', 'Escape'];
+
 const RangeInput = forwardRef(
   (
-    { a11yTitle, name, onChange, onFocus, onBlur, value: valueProp, ...rest },
+    {
+      a11yTitle,
+      name,
+      onChange,
+      onFocus,
+      onBlur,
+      min = 0,
+      max = 100,
+      step = 1,
+      inputValue = false,
+      value: valueProp,
+      ...rest
+    },
     ref,
   ) => {
     const formContext = useContext(FormContext);
 
     const [value, setValue] = formContext.useFormInput(name, valueProp);
 
     const [focus, setFocus] = useState();
+
+    const [showInputValue, setShowInputValue] = useState(false);
+
+    const checkInputAndChange = event => {
+      const newEvent = event;
+      if (newEvent.target.value < min || !newEvent.target.value) {
+        newEvent.target.value = min;
+      } else if (newEvent.target.value > max) {
+        newEvent.target.value = max;
+      }
+      setValue(newEvent.target.value);
+      if (onChange) onChange(newEvent);
+    };
+
     return (
-      <StyledRangeInput
-        aria-label={a11yTitle}
-        ref={ref}
-        name={name}
-        focus={focus}
-        value={value}
-        {...rest}
-        onFocus={event => {
-          setFocus(true);
-          if (onFocus) onFocus(event);
-        }}
-        onBlur={event => {
-          setFocus(false);
-          if (onBlur) onBlur(event);
-        }}
-        onChange={event => {
-          setValue(event.target.value);
-          if (onChange) onChange(event);
-        }}
-        type="range"
-      />
+      <>
+        <StyledRangeInput
+          aria-label={a11yTitle}
+          ref={ref}
+          name={name}
+          focus={focus}
+          value={value}
+          min={min}
+          max={max}
+          step={step}
+          {...rest}
+          onFocus={event => {
+            setFocus(true);
+            if (onFocus) onFocus(event);
+          }}
+          onBlur={event => {
+            setFocus(false);
+            if (onBlur) onBlur(event);
+          }}
+          onChange={event => {
+            setValue(event.target.value);
+            if (onChange) onChange(event);
+          }}
+          onKeyDown={event => {
```

Comment:
This listener fn becomes a little bit bigger, I would suggest extract it to a `const`.
In the other listeners(onBlur, onFocus, onChange) you are able to pass an extra func.
What about add this:

```ts
if (onKeyDown) onKeyDown(event);
```

Reply:
Hello @abnersajr! Thanks for your feedback, I do not used the `if (onKeyDown) onKeyDown(event);` cause I wanted to keep internal this keydown behavior on RangeInput component.

Reply:
`onKeyDown` it's already a value event handler. Some projects might be using it, and remove this can break something.